subprojects/plugins/src/main/java/org/gradle/api/plugins/JvmPluginExtension.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.gradle.api.plugins;
+
+import org.gradle.api.Action;
+import org.gradle.api.Incubating;
+import org.gradle.api.artifacts.Configuration;
+import org.gradle.api.plugins.jvm.JvmVariantBuilder;
+import org.gradle.api.plugins.jvm.JvmEcosystemUtilities;
+import org.gradle.api.plugins.jvm.OutgoingElementsBuilder;
+import org.gradle.api.plugins.jvm.ResolvableConfigurationBuilder;
+
+/**
+ * This extension is shared by JVM plugins (Java, Groovy, Scala, ...)
+ * and can be used to configure JVM specific behavior.
+ *
+ * @since 6.7
+ */
+@Incubating
+public interface JvmPluginExtension {
+    /**
+     * Provides access to the several handy JVM related utilities.
+     */
+    JvmEcosystemUtilities getUtilities();
+
+    /**
+     * Creates an outgoing configuration and configures it with reasonable defaults.
+     * @param name the name of the outgoing configurtion
+     * @param configuration the configuration builder, used to describe what the configuration is used for
+     * @return an outgoing (consumable) configuration
+     */
+    Configuration createOutgoingElements(String name, Action<? super OutgoingElementsBuilder> configuration);
+
+    /**
+     * Creates a configuration which can be used to resolve dependencies for the JVM
+     * ecosystem. It will also create the configurations used to declare dependencies,
+     * also known as "bucket" configurations.
```

Comment:
```suggestion
     * ecosystem. It may also be used to create configuration for declaring dependencies,
     * also known as "bucket" configurations (see ResolvableConfigurationBuilder.usingDependencyBucket).
```